task_02/novikov_dmitry/task_02.cpp

```diff
@@ -0,0 +1,242 @@
+        return this->vertex_id != v.vertex_id;
+    }
+
+    std::string vertex_to_str() const{
+        std::stringstream ss_out;
+        ss_out << this->vertex_id;
+        return ss_out.str();
+    }
+    void get_Info() const {
+        std::string str = std::string(4, ' ');
+        std::cout << "Vertex {" << std::endl; 
+        std::cout << str << "int vertex_id;" << std::endl;
+        std::cout << str << "Vertex ver1;" << std::endl;
+        std::cout << str << "Vertex ver2;" << std::endl;
+        std::cout << "}";
+    }
+    std::tuple<int> get_all_fields() const {
+        return std::make_tuple(vertex_id);
+    }
+};
+
+class Edge {
+private:
+    Vertex ver1;
+    int edge_id;
+    Vertex ver2;
+public:
+    ////other information
+    Edge(int v1_id, int e_id, int v2_id) :   
+        ver1(v1_id), 
+        edge_id(e_id), 
+        ver2(v2_id) {}
+    Edge(const Vertex &v1, int e_id, const Vertex &v2) :         
+        ver1(v1), 
+        edge_id(e_id),
+        ver2(v2) {}
+    Edge(const Edge &item) : 
+        ver1(item.ver1), 
+        edge_id(item.edge_id),
+        ver2(item.ver2) {}
+    //edge(int id, ...) {}
+    bool operator >(const Edge &edge) const {
+        return this->edge_id > edge.edge_id;
+    }
+    bool operator <(const Edge &edge) const {
+        return this->edge_id < edge.edge_id;
+    }
+    bool operator ==(const Edge &edge) const {
+        return this->edge_id == edge.edge_id;
+    }
+    bool operator !=(const Edge &edge) const {
+        return this->edge_id != edge.edge_id;
+    }
+
+    std::string edge_to_str() const{
+        std::stringstream ss_out;
+        ss_out << "{";
+        ss_out << ver1.vertex_to_str();
+        ss_out << ", " << this->edge_id << ", ";
+        ss_out << ver2.vertex_to_str();
+        ss_out << "}";
+        return ss_out.str();
+    }
+    int get_id() const {
+        return this->edge_id;
+    }
+    std::tuple<const Vertex&, int, const Vertex&> 
+        get_all_fields() const {
+        return std::make_tuple(std::ref(ver1), edge_id, std::ref(ver2));
+    }
+    void get_Info() const {
+        std::string str = std::string(4, ' ');
+        std::cout << "Edge {" << std::endl; 
+        std::cout << str << "Vertex ver1;" << std::endl;
+        std::cout << str << "int edge_id;" << std::endl;
+        std::cout << str << "Vertex ver2;" << std::endl;
+        std::cout << "}";
+    }
+};
+
+class Graph {
+public:
+    //new type map Vertices
+    using Edge_ptr_vector = std::vector<std::unique_ptr<Edge>>;
+    std::map<Vertex, Edge_ptr_vector> vertex_map;
+    //new type map for Edges
+    using Edge_ptr_map = std::map<int, std::unique_ptr<Edge>>;
+    Edge_ptr_map edge_map;
+
+    Graph(const Vertex &v) {
+        add_vertex(v);
+    }
+    Graph(const Edge &edge) {
+        add_edge(edge);
+    }
+    template <typename T>
+    Graph(const T &vec) {
+        auto it = vec.begin();
+        for (; it != vec.end(); ++it) {

```

Comment:
Принято такое объявление цикла `for`
```suggestion
        for (auto it = vec.begin(); it != vec.end(); ++it) {
```